### ISSUE TITLE
Dismiss action to the Sheet component

### ISSUE THREAD

**Issue by andredelramo-hm** [Telefonica/mistica-design#2384]
## Description
Add a visible dismiss action to the Sheet component to ensure all users can close the view accessibly, without relying solely on gestures or outside tap.

In addition, the dismiss action should be aligned with the title on both desktop and mobile.

## Tasks

- [x] [Update specs](https://www.figma.com/design/nxToAYk7FGC86TS0mY61TU/%F0%9F%94%B8-Sheet-Specs?node-id=2-61)
- [ ] Design documentation
- [x] Add to mobile library
- [x] Add to desktop library
- [ ] Add to UI Kit
- [x] [Android ticket](https://jira.tid.es/browse/ANDROID-17173)
- [x] [iOS ticket](https://jira.tid.es/browse/IOS-12092)
- [x] [Web ticket](https://jira.tid.es/browse/WEB-2353)
